src/v/compat/id_allocator_generator.h

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2022 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+#pragma once
+
+#include "cluster/errc.h"
+#include "cluster/id_allocator.h"
+#include "cluster/types.h"
+#include "compat/generator.h"
+#include "model/timeout_clock.h"
+#include "random/generators.h"
+
+#include <chrono>
+#include <cstdint>
+#include <limits>
+#include <type_traits>
+
+namespace compat {
+
+template<>
+struct instance_generator<cluster::allocate_id_request> {
+    static cluster::allocate_id_request random() {
+        return cluster::allocate_id_request{model::timeout_clock::duration(
+          random_generators::get_int<int64_t>(-100000, 100000) * 1000000)};
+    }
+
+    static std::vector<cluster::allocate_id_request> limits() {
+        return {
+          cluster::allocate_id_request{model::timeout_clock::duration(0)},
+          /* ------ BUGGGG --- overflow in duration_cast to nanoseconds--- */
```

Comment:
@dotnwat FYI.. timeout_clock::duration --> nanoseconds.
(sorry for the duplicate comment, had to do a force push to rewrite the commits, older comment/code are gc-ed).